server/src/test/scala/org/http4s/server/middleware/CSRFSpec.scala

```diff
@@ -0,0 +1,120 @@
+package org.http4s.server.middleware
+
+import org.http4s._
+import org.http4s.dsl._
+import org.http4s.headers.{`Set-Cookie` => HCookie}
+import fs2.Task
+import org.http4s.implicits._
+import cats.implicits._
+
+class CSRFSpec extends Http4sSpec {
+
+  val dummyService: HttpService = HttpService {
+    case GET -> Root =>
+      Thread.sleep(1) //Fix so clock doesn't compute the same nonce, to emulate a real service
```

Comment:
I didn't catch this the first time but we can use fs2 primtives to sleep without blocking a thread.  Not that blocking a thread for a millisecond particularly matters, but because it's a bad example for others who throw larger constants into this.